packages/react-events/src/Focus.js

```diff
@@ -224,11 +291,13 @@ const FocusResponder = {
       focusTarget: null,
       isFocused: false,
       isLocalFocusVisible: false,
+      isFocusWithin: false,
+      isFocusWithinVisible: false,
       pointerType: '',
     };
   },
   allowMultipleHostChildren: false,
-  stopLocalPropagation: true,
+  stopLocalPropagation: false,
```

Comment:
This shouldn't be necessary

Reply:
The problem was if you had nested `<Focus>` components. Then focus within wouldn't work on the outer one since event propagation was stopped at the inner one.

Reply:
That's the desired behaviour here. We don't want <Focus> to propagate ideally.

Reply:
Normal focus events aren't triggered on parents because of the check that the event target equals the immediate child of the focus component. But with focus within, I presume it should work like the `:focus-within` CSS selector, which is true when any descendant is focused. If there are nested `<Focus>` components, that would break the way focus within works and prevent focus within from being triggered since events are stopped at the inner most <Focus>.

Reply:
That makes sense, but you're changing the rest of how `<Focus>` works by changing `stopLocalPropagation`. You could make `stopLocalPropagation` `false`, but then you'd need to add additional logic into the responder callbacks to properly ensure local propagation is stopped. `FocusScope` actually gives an example of doing this by taking over responder ownership via `requestResponderOwnership`, which would block other event components of the same responder type from being propagated to.

Reply:
Ok... I can look. Should there be tests for that behavior? I assumed that since no tests failed when changing that setting that it was still working correctly. Also assumed it was just `event.stopPropagation()` but I guess that's also not true.

Reply:
`event.stopPropagation` doesn't mean anything to React as we don't use DOM propagation. We should add more test coverage to `Focus` as it's a bit lacking.

Reply:
Focus has logic that ignores events if they aren't dispatched on the currentTarget. So I don't think there's missing test coverage or that managing ownership will be necessary. Changing local propagation would only make a difference for focus-within, it's just about which option makes the most sense.

For example, Hover intentionally doesn't pierce event component boundaries even though you might think it should at first. I mentioned in the prior discussion that we could in principle support both behaviours for focus-within, without altering the behaviour of any other callbacks either way. I'm sympathetic to the behaviour you chose to implement here by default

I'm not feeling well today but will test this PR out internally soon. Thanks!

Reply:
@necolas The issue is that if you nest `<Focus>` event components with `stopLocalPropagation` set to `false:

```jsx
<Focus>
  <div>
    <Focus>
      <span>
```
Then trigger a focus on the `<span>`, then inner `<div>` will be the `currentTarget`, not the outer `<span>`.

Reply:
The behaviour of everything we've currently implemented should remain unchanged.

Reply:
So do you recommend following @trueadm's advice above about using responder ownership similar to `FocusScope` to implement this? How do I verify that it's working?

Reply:
I don't think we should change this. Let's leave this the same as it is for now. I want to think more on how nested <Focus> and other components should be handled, but I don't think that work should be coupled with this PR.